### ISSUE TITLE
Missing key ['pc_indices'] in the dataset. 

### ISSUE THREAD

**Issue by yzheng97** [LiuShihHung/GICN#3]
Thanks for your work. I notice that the key 'pc_indices' appears both in train and inference script, but missing in the h5 files.

**Comment by escanillans** [comment]
also getting the same error. @yzheng97 were you able to figure out what the problem is?

`KeyError: "Unable to open object (object 'pc_indices' doesn't exist)"`

**Comment by FishWantToFly** [comment]
We haven't organized codes about generating dataset including **key ['pc_indices']** yet. 
In actual, we just additionally record a new column **pc_indices** representing the index of this point in raw point cloud when generating cubes from raw point cloud.

**Comment by escanillans** [comment]
@FishWantToFly So in the source code - `pc_indices = fin['pc_indices'][block_id] # [4096]` just represents an array from [0,...,4095]?

**Comment by FishWantToFly** [comment]
No, it actually represents the original point cloud index in raw point cloud, for sampled 4096 points (because we sample 4096 points per block)

**Comment by GdMacmillan** [comment]
Hi @FishWantToFly @LiuShihHung  I work with @escanillans and we are trying to prove your method on some indoor scene data. I believe it would be best if we can generate validation resulting from your pre-trained model. Can you possible attache or link to a copy of the dataset with the _pc_indices_ or the script used to generate these?

**Comment by meidachen** [comment]
Same problem about the pc_indices, any help would be greatly appreciated!

**Comment by ammaradel** [comment]
```pc_indices``` isn't used in the code anyway. I just commented the line.

**Comment by yzheng97** [comment]
> `pc_indices` isn't used in the code anyway. I just commented the line.

Then how do you run main_eval.py ?

**Comment by ammaradel** [comment]
Sorry, I was talking about ```train.py```. I didn't run ```main_eval.py``` yet. 
The h5 files are loaded successfully but I still have a _"CUDA kernel failed"_ error message.

**Comment by ammaradel** [comment]
I don't know if ```pc_indices``` is really necessary. I defined it as: ```pc_indices = np.arange(4096)``` [here](https://github.com/LiuShihHung/GICN/blob/29cd2975e539cf09e270159934d4ca3a7da01f0a/helper_data_s3dis.py#L71), fixed some issues about squeezing numpy arrays, and ```main_eval.py``` finally worked.

**Comment by PengboLi1998** [comment]
Hi, @ammaradel .Could you please tell me how do you fix the issues about squeezing numpy arrays.I also meet the same error.
